src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -10995,7 +10995,7 @@ protected override void VisitInterpolatedStringHandlerConstructor(BoundExpressio
 
         public override BoundNode? VisitDiscardExpression(BoundDiscardExpression node)
         {
-            var result = TypeWithAnnotations.Create(node.Type);
+            var result = TypeWithAnnotations.Create(node.Type, node.NullableAnnotation);
```

Comment:
Just to confirm, do we always get here with oblivious for `out _` scenario?

Reply:
AFAIK all method call arguments are visited in `NullableWallker.cs`. See [code](https://github.com/dotnet/roslyn/blob/795612c361a8d5d9efde45f5f00da214e39a9659/src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs#L6398). The `out T/var/ _` parameters are visited by the method above.